File: Python/Lecture_07_Lab_Exercise_1_Laplace.py
"""
========================================================================
        ╦ ╦┌─┐┌─┐┌─┐┌─┐┌┬┐  ╔╦╗┌─┐┌─┐┌┬┐┬ ┬  ╔╗ ┌─┐┬  ┌─┐┬ ┬┌─┐
        ╠═╣│ │└─┐└─┐├─┤│││  ║║║├─┤│ ┬ ││└┬┘  ╠╩╗├─┤│  ├─┤├─┤├─┤
        ╩ ╩└─┘└─┘└─┘┴ ┴┴ ┴  ╩ ╩┴ ┴└─┘─┴┘ ┴   ╚═╝┴ ┴┴─┘┴ ┴┴ ┴┴ ┴
========================================================================
# Author: Hossam Magdy Balaha
# Permissions and Citation: Refer to the README file.
"""

# Import necessary libraries.
from sympy import symbols, exp, sin, cos, laplace_transform

# Define the different symbols.
t, s = symbols("t s", real=True)
a, b, c, d = symbols("a b c d", real=True, positive=True)

# Laplace Transform of a function.

# Example 1: Laplace Transform of f(t) = t * e^(-2t).
f1 = t * exp(-2 * t)
F1 = laplace_transform(f1, t, s)
# Display the result of the Laplace Transform.
print(f"Laplace Transform of {f1} is:\n\t{F1[0]}")
print("")

# Example 2: Laplace Transform of f(t) = e^(-2t) * sin(3t).
f2 = exp(-2 * t) * sin(3 * t)
F2 = laplace_transform(f2, t, s)
# Display the result of the Laplace Transform.
print(f"Laplace Transform of {f2} is:\n\t{F2[0]}")
print("")

# Example 3: Laplace Transform of f(t) = 1 - e^(-at) * (b * sin(ct) + d * cos(ct)).
f3 = 1 - exp(-a * t) * (b * sin(c * t) + d * cos(c * t))
F3 = laplace_transform(f3, t, s)
# Display the result of the Laplace Transform.
print(f"Laplace Transform of {f3} is:\n\t{F3[0]}")

File: MATLAB/Lecture_07_Lab_Exercise_1_Laplace.m
% ===========================================================================
%         ╦ ╦┌─┐┌─┐┌─┐┌─┐┌┬┐  ╔╦╗┌─┐┌─┐┌┬┐┬ ┬  ╔╗ ┌─┐┬  ┌─┐┬ ┬┌─┐
%         ╠═╣│ │└─┐└─┐├─┤│││  ║║║├─┤│ ┬ ││└┬┘  ╠╩╗├─┤│  ├─┤├─┤├─┤
%         ╩ ╩└─┘└─┘└─┘┴ ┴┴ ┴  ╩ ╩┴ ┴└─┘─┴┘ ┴   ╚═╝┴ ┴┴─┘┴ ┴┴ ┴┴ ┴
% ===========================================================================
%
% Author: Hossam Magdy Balaha
% Initial Creation Date: June 26th, 2025
% Last Modification Date: June 26th, 2025
% Permissions and Citation: Refer to the README file.

% Define symbolic variables.
syms t s a b c d real;

% Example 1: Laplace Transform of f(t) = t * exp(-2*t).
f1 = t * exp(-2 * t);
F1 = laplace(f1, t, s);

% Display result.
disp('=== Laplace Transform of f(t) = t * exp(-2*t) ===')
disp(['f(t) = ', char(f1)]);
disp(['F(s) = ', char(F1)]);
disp('');

% Example 2: Laplace Transform of f(t) = exp(-2t) * sin(3t).
f2 = exp(-2 * t) * sin(3 * t);
F2 = laplace(f2, t, s);

% Display result.
disp('=== Laplace Transform of f(t) = exp(-2t)*sin(3t) ===')
disp(['f(t) = ', char(f2)]);
disp(['F(s) = ', char(F2)]);
disp('');

% Example 3: Laplace Transform of f(t) = 1 - exp(-a*t)*(b*sin(c*t) +
% d*cos(c*t)).
f3 = 1 - exp(-a*t) * (b*sin(c*t) + d*cos(c*t));
F3 = laplace(f3, t, s);

% Display result.
disp('=== Laplace Transform of f(t) = 1 - exp(-a*t)*(b*sin(c*t) + d*cos(c*t)) ===')
disp(['f(t) = ', char(f3)]);
disp(['F(s) = ', char(F3)]);